pennylane/ops/qubit/parametric_ops_multi_qubit.py

```diff
@@ -258,23 +284,44 @@ class PauliRot(Operation):
         "Z": np.array([[1, 0], [0, 1]]),
     }
 
+    def _check_word_batching(self, pauli_word):
+        """Check that the broadcasting/batching of a Pauli word is valid and
+        consistent with potential broadcasting of a ``PauliRot`` instance."""
+        invalid_word_msg = (
+            f"The given Pauli word '{pauli_word}' contains characters that are not "
+            "allowed. Allowed characters are I, X, Y and Z"
+        )
+        num_wires = len(self.wires)
+        invalid_len_msg = lambda length: (
+            f"The given Pauli word has length {length}, length "
+            f"{num_wires} was expected for wires {self.wires}"
+        )
+        if isinstance(pauli_word, str):
+            # Single string, no broadcasting
+            if not PauliRot._check_pauli_word(pauli_word):
+                raise ValueError(invalid_word_msg)
+            if not (_len := len(pauli_word)) == num_wires:
+                raise ValueError(invalid_len_msg(_len))
+        else:
+            # Iterable of strings, broadcasting
+            if self._batch_size and self._batch_size != len(pauli_word):
+                raise ValueError(
+                    "When broadcasting the rotation angle and the Pauli word, the broadcasting "
+                    f"dimensions have to match, but got {self._batch_size} and {len(pauli_word)}."
+                )
+
+            if not PauliRot._check_pauli_word("".join(pauli_word)):
+                raise ValueError(invalid_word_msg)
+            if not all((_len := len(w)) == num_wires for w in pauli_word):
+                raise ValueError(invalid_len_msg(_len))
+
+            self._batch_size = len(pauli_word)
+
     def __init__(self, theta, pauli_word, wires=None, id=None):
         super().__init__(theta, wires=wires, id=id)
         self.hyperparameters["pauli_word"] = pauli_word
```

Comment:
If we are allowing `pauli_word` to be an interable, we will need to makes sure that its a hashable iterable (aka tuple). We need to make sure the metadata returned by `op._flatten()` stays hashable.

Reply:
Do I understand correctly that doing `pauli_word = pauli_word if isinstance(pauli_word, str) else tuple(pauli_word)` is sufficient to accomplish this? In this case I think we do not even need a custom `_flatten` method :)

NB: In `Operation._flatten`, the tuple comprehension over `hyperparameters.items()` could be replaced by `tuple(self.hyperparameters.items())`, I think? :thinking:

Reply:
Casting to tuple should be sufficient.  

Can yes, `tuple(self.hyperparameters.items())` may work.